.github/workflows/e2e-main.yml

```diff
@@ -1,11 +1,8 @@
-name: E2E
+name: E2E - Main
 on:
   push:
     branches:
```

Comment:
do we also want this job to be run after every PR merge? if not, should we remove the on / push?

Reply:
I discussed with @MuckT and came up with the following - he may have other insights as well.

Run against multiple OS versions when:
 - A commit is pushed to main (so, _after_ a PR is merged)
 - When called from the Github UI
 - On a schedule, once per day

Run against a single OS version when:
 - A PR is opened, before it can be merged

This ensures that PRs can pass checks relatively quickly, but we still test against the full suite of OS versions on `main`. We will learn about potential OS version incompatibilities as we merge into main, without having to wait for a full regression test. We can potentially learn about service outages/breaking changes with the daily run as well.

I feel like what is in this strategy is a pretty good balance. In general if a PR has passed E2E tests against a specific OS version, we're probably good to move forward. The daily/push-to-main tests are just additional checks against all the other OS versions. It would be nice to know about OS-specific issues before we cut a release and test against it - and at the same time, if we find tests to be flaky, it's ok to just re-run tests/ignore a single failed run on `main`.

I'm happy to make changes if you see any issues or have suggestions for improvements. Updating workflow files pretty straightforward.